src/GoogleAdWordsEventForwarder.js

```diff
@@ -124,6 +126,75 @@
             return 'Can\'t send to forwarder ' + name + ', not initialized. Event added to queue.';
         }
 
+        function buildNestedEnhancedConversionData(customFlags) {
+            const sanitizedConversionData = {};
+            const conversionData = customFlags[ENHANCED_CONVERSION_DATA];
+
+            if (conversionData.email) {
+                sanitizedConversionData.email = conversionData.email;
+            }
+            if (conversionData.phone_number) {
+                sanitizedConversionData.phone_number = conversionData.phone_number;
+            }
+            if (conversionData.first_name) {
+                sanitizedConversionData.first_name = conversionData.first_name;
+            }
+            if (conversionData.last_name) {
+                sanitizedConversionData.last_name = conversionData.last_name;
+            }
+            if (conversionData.home_address) {
+                sanitizedConversionData.home_address = conversionData.home_address;
+            }
+
+            return sanitizedConversionData;
+        }
+
+        function buildStringEnhancedConversionData(customFlags) {
+            const sanitizedConversionData = {};
+
+            if(customFlags[ENHANCED_CONVERSION_DATA + '.email']) {
+                sanitizedConversionData.email = customFlags[ENHANCED_CONVERSION_DATA + '.email'];
+            }
+
+            if(customFlags[ENHANCED_CONVERSION_DATA + '.phone_number']) {
+                sanitizedConversionData.phone_number = customFlags[ENHANCED_CONVERSION_DATA + '.phone_number'];
+            }
+
+            if(customFlags[ENHANCED_CONVERSION_DATA + '.first_name']) {
+                sanitizedConversionData.first_name = customFlags[ENHANCED_CONVERSION_DATA + '.first_name'];
+            }
+
+            if(customFlags[ENHANCED_CONVERSION_DATA + '.last_name']) {
+                sanitizedConversionData.last_name = customFlags[ENHANCED_CONVERSION_DATA + '.last_name'];
+            }
+
+            if(Object.keys(customFlags).toString().includes('.home_address')) {
```

Comment:
this feels a little sketchy - it'd match stuff that wouldn't necessarily be what we're looking for. I think you could delete it and the code would do the same but be more readable.